Core/src/Geometry/CuboidVolumeStack.cpp

```diff
@@ -0,0 +1,893 @@
+          b.updatedBounds->set(dirBoundIdx, bHlNew);
+          break;
+        }
+        case AttachmentStrategy::Gap: {
+          ACTS_VERBOSE(" -> Strategy: Create a gap volume");
+          double gapHl = (b.min(m_dir) - a.max(m_dir)) / 2.0;
+          double gapMid = (b.min(m_dir) + a.max(m_dir)) / 2.0;
+
+          ACTS_VERBOSE("  - Gap half length: " << gapHl << " at "
+                                               << axisDirectionName(m_dir)
+                                               << ": " << gapMid);
+
+          Translation3 gapTranslation(Vector3::Unit(dirIdx) * gapMid);
+
+          double min1 = std::min(a.min(m_dirOrth1), b.min(m_dirOrth1));
+          double max1 = std::max(a.max(m_dirOrth1), b.max(m_dirOrth1));
+
+          double min2 = std::min(a.min(m_dirOrth2), b.min(m_dirOrth2));
+          double max2 = std::max(a.max(m_dirOrth2), b.max(m_dirOrth2));
+
+          Transform3 gapLocalTransform{gapTranslation};
+          Transform3 gapGlobalTransform = m_groupTransform * gapLocalTransform;
+
+          std::array<double, 3> gapBoundsVals;
+          gapBoundsVals[dirIdx] = gapHl;
+          gapBoundsVals[static_cast<std::size_t>(m_dirOrth1)] =
+              (max1 - min1) / 2;
+          gapBoundsVals[static_cast<std::size_t>(m_dirOrth2)] =
+              (max2 - min2) / 2;
+          auto gapBounds = std::make_shared<CuboidVolumeBounds>(gapBoundsVals);
+          auto gap = addGapVolume(gapGlobalTransform, gapBounds);
+          gapVolumes.emplace_back(*gap, m_groupTransform);
+
+          break;
+        }
+        default:
+          ACTS_ERROR("Attachment strategy " << strategy << " not implemented");
+          std::stringstream ss;
+          ss << strategy;
+          throw std::invalid_argument("Attachment strategy " + ss.str() +
+                                      " not implemented");
+      }
+    }
+  }
+
+  return gapVolumes;
+}
+
+void CuboidVolumeStack::printVolumeSequence(
+    const std::vector<VolumeTuple>& volumes, const Logger& logger,
+    Acts::Logging::Level lvl) {
+  if (!logger().doPrint(lvl)) {
+    return;
+  }
+  for (const auto& vt : volumes) {
+    std::stringstream ss;
+    ss << std::fixed;
+    ss << std::setprecision(3);
+    ss << std::setfill(' ');
+
+    int w = 9;
+
+    for (const auto& axis :
+         {AxisDirection::AxisX, AxisDirection::AxisY, AxisDirection::AxisZ}) {
+      ss << axisDirectionName(axis) << ": [ " << std::setw(w) << vt.min(axis)
+         << " <- " << std::setw(w) << vt.mid(axis) << " -> " << std::setw(w)
+         << vt.max(axis) << " ]\n";
+    }
+    logger().log(lvl, ss.str());
+  }
+}
+
+void CuboidVolumeStack::checkVolumeAlignment(
+    const std::vector<VolumeTuple>& volumes, const Logger& logger) const {
+  std::size_t n = 0;
+  auto dirIdx = static_cast<std::size_t>(m_dir);
+  auto dirOrth1Idx = static_cast<std::size_t>(m_dirOrth1);
+  auto dirOrth2Idx = static_cast<std::size_t>(m_dirOrth2);
```

Comment:
Same comment applies here. I think the best approach is to have a (hidden) function that converts between these in a safe way with a `switch`, or maybe a member method that returns the orthogonal axes as native Eigen axes (`0, 1, 2`).